### PR TITLE
MCR-2359 (2) migrate PI to configurable instances

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2359).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 1c30ed6f0793fd27fb6b3613e4575d8ee28f7404! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6936-pull_request-1226

File: mycore-pi/src/main/java/org/mycore/pi/MCRPIService.java [hunk 1]
```diff
@@ -205,21 +197,14 @@ protected void checkConfiguration() throws MCRConfigurationException {
             .orElseThrow(generatorPropertiesNotSetError);
 
         String generatorPropertyKey = GENERATOR_CONFIG_PREFIX + generatorName;
-        String className = MCRConfiguration2.getStringOrThrow(generatorPropertyKey);
+        Optional<MCRPIGenerator<T>> generatorOptional = MCRConfiguration2.getInstanceOf(generatorPropertyKey);
```
Comment (rsteph-de) [inline mycore-pi/src/main/java/org/mycore/pi/MCRPIService.java]: orElseThrow(...)

File: mycore-pi/src/main/java/org/mycore/pi/doi/cli/MCRDOICommands.java [hunk 2]
```diff
@@ -123,7 +126,8 @@ public static void repairIncompleteRegisteredDOI(String doiString, String servic
             + "It also updates all media files. The Service ID{0} is the id from the configuration.",
         order = 10)
     public static void synchronizeDatabase(String serviceID) {
-        MCRDOIService registrationService = new MCRDOIService(serviceID);
+        MCRDOIService registrationService = new MCRDOIService();
+        registrationService.init(serviceID);
```
Comment (rsteph-de) [inline mycore-pi/src/main/java/org/mycore/pi/doi/cli/MCRDOICommands.java:131]: Warum nicht auch hier:
 (MCRDOIService) MCRConfiguration2
                .getInstanceOf(MCRPIServiceManager.REGISTRATION_SERVICE_CONFIG_PREFIX + serviceID).get();

**Review by rsteph-de — CHANGES_REQUESTED** [review]
only some very little improvements suggested

File: mycore-pi/src/main/java/org/mycore/pi/MCRPIService.java [hunk 3]
```diff
@@ -167,33 +172,20 @@ protected void checkConfiguration() throws MCRConfigurationException {
...
-        } catch (NoSuchMethodException e) {
-            throw new MCRConfigurationException(
-                "Configurated class (" + (METADATA_SERVICE_CONFIG_PREFIX + metadataManager)
-                    + ") needs a string constructor: " + className);
-        } catch (IllegalAccessException | InstantiationException | InvocationTargetException e) {
-            throw new MCRException(e);
+        String msProperty = METADATA_SERVICE_CONFIG_PREFIX + metadataService;
+        Optional<MCRPIMetadataService<T>> metadataServiceOptional = MCRConfiguration2.getInstanceOf(msProperty);
+        if(metadataServiceOptional.isEmpty()){
+            throw new MCRConfigurationException(msProperty + " is not set or invalid!");
```
Comment (rsteph-de) [inline mycore-pi/src/main/java/org/mycore/pi/MCRPIService.java]: orElseThrow(...)

**Review by rsteph-de — COMMENTED** [review]
one more improvement

**Review by rsteph-de — APPROVED** [review]
looks good for me

**Pull request merged**